packages/backend/src/links/links.service.ts

```diff
@@ -1,45 +1,64 @@
-import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
-import { CreateLinkDto } from './dto/create-link.dto';
-import { UpdateLinkDto } from './dto/update-link.dto';
-import { PrismaService } from 'nestjs-prisma';
-import { Prisma } from '@prisma/client';
+import { BadRequestException, HttpException, Injectable, NotFoundException } from '@nestjs/common'
+import { CreateLinkDto } from './dto/create-link.dto'
+import { UpdateLinkDto } from './dto/update-link.dto'
+import { PrismaService } from 'nestjs-prisma'
+import { Prisma } from '@prisma/client'
 
 @Injectable()
 export class LinksService {
   constructor(private readonly prisma: PrismaService) {}
   async create(createLinkDto: CreateLinkDto) {
     try {
-      return await this.prisma.link.create({ data: createLinkDto });
+      if (this.checkUrl(createLinkDto.url)) {
+        return await this.prisma.link.create({ data: createLinkDto })
+      } else {
+        throw new HttpException('The url you entered is not found!', 404)
+      }
     } catch (e) {
       if (e instanceof Prisma.PrismaClientKnownRequestError) {
         if (e.code === 'P2002') {
-          throw new BadRequestException('This slug violates the unique constraint!');
+          throw new BadRequestException('This slug violates the unique constraint!')
         }
       }
     }
   }
 
   findAll() {
-    return this.prisma.link.findMany();
+    return this.prisma.link.findMany()
   }
 
   async findOne(id: string) {
     const link = await this.prisma.link.findUnique({
       where: { id },
-    });
-
+    })
     if (link === null) {
-      throw new NotFoundException('The link you entered is not found!');
+      throw new NotFoundException('The link you entered is not found!')
     }
-
-    return link;
+    return link
   }
 
   async update(id: string, updateLinkDto: UpdateLinkDto) {
-    return await this.prisma.link.update({ where: { id }, data: updateLinkDto });
+    return await this.prisma.link.update({ where: { id }, data: updateLinkDto })
   }
 
   async remove(id: string) {
-    return await this.prisma.link.delete({ where: { id } });
+    try {
+      return await this.prisma.link.delete({ where: { id } })
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2002') {
+          throw new BadRequestException('This slug violates the unique constraint!')
+        }
+      }
+    }
+  }
```

Comment:
This was meant for the update method, right? No point of checking this on the delete method

Reply:
Yes! I copy pasted to the wrong place, my bad. Will fix